src/mlpack/tests/det_test.cpp

```diff
@@ -494,4 +494,124 @@ BOOST_AUTO_TEST_CASE(TestPrintLeafMembership)
 }
 */
 
+// Test the copy constructor and the copy operator.
+BOOST_AUTO_TEST_CASE(CopyConstructorAndOperatorTest)
+{
+  arma::mat testData(3, 5);
+  
+  testData << 4 << 5 << 7 << 3 << 5 << arma::endr
+           << 5 << 0 << 1 << 7 << 1 << arma::endr
+           << 5 << 6 << 7 << 1 << 8 << arma::endr;
+
+  // Constructing a DTree
+  DTree<arma::mat>* tree = new DTree<arma::mat>(testData);
+  
+  // Constructing copy of DTree using copy constructor and copy operator
+  DTree<arma::mat> tree2(*tree);
+  DTree<arma::mat> tree3 = *tree;
+
+  // Pointers to children of original tree
+  DTree<arma::mat>* leftChild = tree->Left();
+  DTree<arma::mat>* rightChild = tree->Right();
+
+  //Deleting original DTree
+  delete tree;
+
+  //Checking values of DTree made using copy constructor.
+  BOOST_REQUIRE_EQUAL(tree2.maxVals[0], 7);
+  BOOST_REQUIRE_EQUAL(tree2.minVals[0], 3);
+  BOOST_REQUIRE_EQUAL(tree2.maxVals[1], 7);
+  BOOST_REQUIRE_EQUAL(tree2.minVals[1], 0);
+  BOOST_REQUIRE_EQUAL(tree2.maxVals[2], 8);
+  BOOST_REQUIRE_EQUAL(tree2.minVals[2], 1);
+
+  //Checking values of DTree made using copy assignment operator
+  BOOST_REQUIRE_EQUAL(tree3.maxVals[0], 7);
+  BOOST_REQUIRE_EQUAL(tree3.minVals[0], 3);
+  BOOST_REQUIRE_EQUAL(tree3.maxVals[1], 7);
+  BOOST_REQUIRE_EQUAL(tree3.minVals[1], 0);
+  BOOST_REQUIRE_EQUAL(tree3.maxVals[2], 8);
+  BOOST_REQUIRE_EQUAL(tree3.minVals[2], 1);
+
+  //Checking children were safely copied
+  BOOST_REQUIRE(tree2.Left()==leftChild);
+  BOOST_REQUIRE(tree2.Right()==rightChild);
+  BOOST_REQUIRE(tree3.Left()==leftChild);
+  BOOST_REQUIRE(tree3.Right()==rightChild);
```

Comment:
At this point `leftChild` and `rightChild` should be invalid pointers; I bet valgrind will call this an invalid memory access and in some cases this test will fail.